crates/bevy_ecs/src/change_detection.rs

```diff
@@ -181,6 +181,32 @@ pub struct ResMut<'a, T: Resource> {
     pub(crate) ticks: Ticks<'a>,
 }
 
+impl<'w, 'a, T: Resource> IntoIterator for &'a ResMut<'w, T>
+where
+    &'a T: IntoIterator,
+{
+    type Item = <&'a T as IntoIterator>::Item;
+
```

Comment:
style nit: remove the extra blank line here. IMO it doesn't make it any more readable. Same with the others.